sklearn/pipeline.py

```diff
@@ -10,23 +10,98 @@
 # License: BSD
 
 from collections import defaultdict
+from abc import ABCMeta, abstractmethod
+import time
 
 import numpy as np
 from scipy import sparse
 
 from .base import clone, TransformerMixin
 from .externals.joblib import Parallel, delayed
 from .externals import six
-from .utils.metaestimators import if_delegate_has_method
-from .utils import Bunch
+from .utils.metaestimators import if_delegate_has_method, _BaseComposition
+from .utils import Bunch, tosequence
 from .utils.validation import check_memory
 
-from .utils.metaestimators import _BaseComposition
 
 __all__ = ['Pipeline', 'FeatureUnion']
 
 
-class Pipeline(_BaseComposition):
+def _pretty_print(step_info):
+    """Helper method to print the information about execution of a particular
+    step of Pipeline / FeatureUnion (if verbosity is enabled). It receives a
+    string having information about current step and prints it in such a way
+    that its length is 70 characters.
+
+    Parameters
+    ----------
+    step_info : str
+        String of form '[ClassName] (step x of y) step_name ... time_elapsed'
+
+    """
+    name, elapsed = step_info.split('...')
+    print('%s%s%s' % (name, '.' * (70 - len(name + elapsed)), elapsed))
+
+
+class _BasePipeline(six.with_metaclass(ABCMeta, _BaseComposition)):
```

Comment:
It seems that it has been badly merged. This class has been removed in favor of `_BaseComposition`.